rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,295 @@
+
+## Value types
+### Trip Key
+For scheduled trips, Glides does not have the same trip ID used by RTR. For added trips, while an ID is provided, it can be the case that an added trip from the Glides perspective is matches a GTFS trip, in which case RTR will us the GTFS trip ID in publishing the information.
+
+In order to reduce message duplication, a Trip Key is used to identify both added and scheduled trips.
+
+| Added Trip | Scheduled Trip |
+| ---------- | -------------- |
+| `{"serviceDate": "2023-01-20", "id": "ADDED-123"}`|`{"serviceDate": "2023-01-20", "scheduledTime": "01:02:30"}`|
+
+### Trip Type
+When adding a trip, there are three kinds of trips which can be added:
+- `"arrival"`: a trip arriving at `stop` at `time`
+- `"departure"`: a trip departing from `stop` at `time`
+- `"roundTrip"`: a trip departing from `stop` at `time` and expected to return to `stop`
+
+### Actor
+Each message will include an `actor` key, indicating who performed the given intervention.
+```json
+{
+  "username": "example",
+  "badgeNumber": "123456"
+}
+
+or
+
+{
+  "username": "pswartz"
+}
+```
+It is represented as an object to provide future extensibility if needed.
+
+### Operator
+Operators will be described by their badge number:
+```json
+{
+  "badgeNumber": "123"
+}
+```
+### Car
+Cars (individual parts of a train consist) will be described by their 4 digit number.
+```json
+{
+  "carNumber": "3800"
+}
+```
+### Optional
+A value that can either be an instance of T, or `null`. For example, Optional Car can be
+```json
+{ "carNumber": "3800"}
+# or 
+null
+```
+## Events
+### Operator signed in
+At the start of their shift, operators need to confirm that they are fit-for-duty and do not have any electronic devices. They do this by either physically signing a paper trainsheet, or by typing their badge number into Glides.
+
+Event type: `com.mbta.ctd.glides.operator-signed-in.v1`
+Fields in the event:
+- `actor` (Actor): the inspector who signed in the operator
+- `operator` (Operator): the operator who signed in
+- `signedInAt` (RFC3999 timestamp): the time at which they signed in (separate from the `time` of the event)
+
+### Trip added
+This creates a new trip in the timesheet. It may or may not be a new trip relative to GTFS (see the splitting a train reference example).
+
+Event type: `com.mbta.ctd.glides.trip-added.v1`
+Fields in the event:
+- `actor` (Actor): the inspector adding the trip
+- `tripKey` (Trip Key): provides a unique identifier for the newly created trip
+- `stop` (string): the GTFS stop ID of the station at which the trip will be departing/arriving
```

Comment:
**question:** would this be the parent stop ID or the specific platform stop ID?

Reply:
I meant for it to be the parent stop ID; I'll make that clearer.